component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordChangeEnforcerOnExpiration.java

```diff
@@ -234,20 +234,38 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                     log.debug("Updated user credentials of " + tenantAwareUsername);
                 }
             } catch (org.wso2.carbon.user.core.UserStoreException e) {
-                if(e.getCause() instanceof PrivilegedActionException) {
+                if(e.getMessage().contains("InvalidOperation")){
                     if (log.isDebugEnabled()) {
-                        log.debug("Credential not valid. Credential must be a non null string with following format, "
-                                + userStoreManager.getRealmConfiguration().getUserStoreProperty("PasswordJavaRegEx"));
+                        log.debug("InvalidOperation Invalid operation. User store is read only.", e);
                     }
-                    String errorMsg = userStoreManager.getRealmConfiguration()
-                            .getUserStoreProperty("PasswordJavaRegExViolationErrorMsg");
-                    if (errorMsg != null) {
-                        throw new AuthenticationFailedException(errorMsg);
+                    throw new AuthenticationFailedException(
+                            "InvalidOperation Invalid operation. User store is read only");
```

Comment:
It is not correct to repeat "InvalidOperation Invalid operation" It should be "Invalid operation. User store is ready only"